aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -335,14 +342,44 @@ impl AptosVM {
         let mut session = self
             .vm_impl
             .new_session(resolver, SessionId::epilogue_meta(txn_data));
-        let fee_statement = AptosVM::fee_statement_from_gas_meter(txn_data, gas_meter, 0);
 
-        match TransactionStatus::from_vm_status(
-            error_code.clone(),
-            self.vm_impl
+        if matches!(transaction_status, TransactionStatus::Keep(_))
+            && txn_data.fee_payer().is_some()
+            && txn_data.sequence_number == 0
```

Comment:
I think account sequence number starts from 0 right? so there could be a valid account with sequence number 0, in that case if the txn fails, I guess it just needs to pay the extra gas for calling create_account_if_does_not_exist

Reply:
it would have to pay that in the happy path anyway...